DataFormats/L1TrackTrigger/src/TTTrack_TrackWord.cc

```diff
@@ -0,0 +1,400 @@
+                                     unsigned int phi0,
+                                     unsigned int eta,
+                                     unsigned int z0,
+                                     unsigned int d0,
+                                     unsigned int theChi2,
+                                     unsigned int theBendChi2,
+                                     unsigned int theHitPattern,
+                                     unsigned int iSpare)
+    : iRinv(theRinv),
+      iphi(phi0),
+      ieta(eta),
+      iz0(z0),
+      id0(d0),
+      ichi2(theChi2),          //revert to other packing?  Or will be unpacked wrong
+      iBendChi2(theBendChi2),  // revert to ogher packing? Or will be unpacked wrong
+      ispare(iSpare),
+      iHitPattern(theHitPattern) {
+  initialize();
+}
+
+// one for already-digitized values:
+
+void TTTrack_TrackWord::setTrackWord(unsigned int theRinv,
+                                     unsigned int phi0,
+                                     unsigned int eta,
+                                     unsigned int z0,
+                                     unsigned int d0,
+                                     unsigned int theChi2,
+                                     unsigned int theBendChi2,
+                                     unsigned int theHitPattern,
+                                     unsigned int iSpare) {
+  iRinv = theRinv;
+  iphi = phi0;
+  ieta = eta;
+  iz0 = z0;
+  id0 = d0;
+  ichi2 = theChi2;          //revert to other packing?  Or will be unpacked wrong
+  iBendChi2 = theBendChi2;  // revert to ogher packing? Or will be unpacked wrong
+  ispare = iSpare;
+  iHitPattern = theHitPattern;
+
+  initialize();
+}
+
+// one for floats:
+void TTTrack_TrackWord::setTrackWord(const GlobalVector& Momentum,
+                                     const GlobalPoint& POCA,
+                                     double theRinv,
+                                     double theChi2,
+                                     double theBendChi2,
+                                     unsigned int theHitPattern,
+                                     unsigned int iSpare) {
+  initialize();
+
+  // first, derive quantities to be packed
+
+  float rPhi = Momentum.phi();  // this needs to be phi relative to center of sector ****
+  float rEta = Momentum.eta();
+  float rZ0 = POCA.z();
+  float rD0 = POCA.perp();
+
+  // bin, convert to integers, and pack
+
+  unsigned int seg1, seg2, seg3;
+  seg1 = 0;
+  seg2 = 0;
+  seg3 = 0;
+
+  //eta
+
+  std::cout << " digitizing eta " << NEtaBits << std::endl;
+  ieta = digitize_Signed(rEta, NEtaBits, 0, valLSBEta);
+
+  //z0
+  iz0 = digitize_Signed(rZ0, NZ0Bits, 0, valLSBZ0);
+
+  //chi2 has non-linear bins
+
+  ichi2 = 0;
+
+  for (unsigned int ibin = 0; ibin < Nchi2; ++ibin) {
+    ichi2 = ibin;
+    if (theChi2 < chi2Bins[ibin])
+      break;
+  }
+  std::cout << "chi2: " << theChi2 << " chi2 bin " << ichi2 << std::endl;
```

Comment:
cout is not allowed, use message logger (everywhere in this file)

Reply:
oops, forgot those were in there